dnf/yum/misc.py

```diff
@@ -373,7 +373,7 @@ def _getloginuid():
         with open("/proc/self/loginuid") as fo:
             data = fo.read()
             return int(data)
-    except (IOError, ValueError):
```

Comment:
I do not think this is correct. When you try to open non-existent file, Python will raise IOError (or its more specific subclass FileNotFoundError in the case of python3). So the `int(data)` line is never executed.

I downloaded and run in qemu the risc image mentioned in the bz and found this:

```
# rpm -q dnf
dnf-2.7.5-8.fc28.noarch

# grep -A11 'def _getloginuid' /usr/lib/python3.6/site-packages/dnf/yum/misc.py 
def _getloginuid():
    """ Get the audit-uid/login-uid, if available. None is returned if there
        was a problem. Note that no caching is done here. """
    #  We might normally call audit.audit_getloginuid(), except that requires
    # importing all of the audit module. And it doesn't work anyway: BZ 518721
    try:
        with open("/proc/self/loginuid") as fo:
            data = fo.read()
            return int(data)
    except (IOError, ValueError):
        return None
```
The version of dnf used on risc is too old and is missing the fallback to os.getuid() introduced by commit https://github.com/rpm-software-management/dnf/commit/70779df36929b5ccd621941a78c162ae3b8ea7c2

Reply:
Thanks a lot for comment. Good point.